ui/tests/helpers/module-for-job.js

```diff
@@ -136,6 +136,20 @@ export default function moduleForJob(
         );
       });
 
+      test('clicking in a task group row navigates to that task group', async function (assert) {
+        const tgRow = JobDetail.taskGroups[0];
+        const tgName = tgRow.name;
+        console.log(tgName);
+
+        await tgRow.visitRow();
+
+        const expectedURL = job.namespace
+          ? `/jobs/${encodeURIComponent(job.name)}@${job.namespace}/${tgName}`
+          : `/jobs/${encodeURIComponent(job.name)}/${tgName}`;
```

Comment:
`question`: why is `encodeURIComponent` necessary here? Also, could we use a different URL-based API like `currentRoute` instead?

Reply:
`encodeURIComponent` is necessary to properly handle jobs that have URL characters in them. For example, a dispatch job would look something like `example/dispatch-1729432782`. That slash is an URI component, so the URL is actually `example%2Fdispatch-1729432782`. Without `encodeURIComponent` the test fails because the strings are not equal.


`currentRoute` is [not available](https://github.com/emberjs/ember-test-helpers/blob/master/API.md#routing-helpers) in the `ember-test-helpers` library we use and `currentRouteName` is not enough since it doesn't provide the route parameters.